rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/interactor/CovidPatientVisitFragmentInteractorTest.java

```diff
@@ -0,0 +1,57 @@
+package io.ona.rdt.robolectric.interactor;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+import org.robolectric.util.ReflectionHelpers;
+import org.smartregister.domain.Event;
+import org.smartregister.domain.db.EventClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.ona.rdt.domain.Visit;
+import io.ona.rdt.interactor.CovidPatientVisitFragmentInteractor;
+import io.ona.rdt.repository.PatientHistoryRepository;
+import io.ona.rdt.robolectric.RobolectricTest;
+
+public class CovidPatientVisitFragmentInteractorTest extends RobolectricTest {
+
+    private CovidPatientVisitFragmentInteractor interactor;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        interactor = new CovidPatientVisitFragmentInteractor();
+    }
+
+    @Test
+    public void testGetPatientVisits() {
+        PatientHistoryRepository patientHistoryRepository = Mockito.mock(PatientHistoryRepository.class);
+        ReflectionHelpers.setField(interactor, "patientHistoryRepository", patientHistoryRepository);
+
+        int listSize = 1;
+        List<EventClient> eventClientList = new ArrayList<>(listSize);
+        Event event = new Event();
+        event.setEventDate(DateTime.now());
+        eventClientList.add(new EventClient(event));
+
+        Mockito.when(patientHistoryRepository.getEventsByUniqueDate(ArgumentMatchers.anyString())).thenReturn(eventClientList);
+
+        List<Visit> visitList = interactor.getPatientVisits("");
+        Assert.assertEquals(listSize, visitList.size());
+        Assert.assertEquals("Visit 1", visitList.get(listSize - 1).getVisitName());
+        Assert.assertNull(visitList.get(listSize - 1).getDateOfVisit());
```

Comment:
Could we verify the actual value here?

Reply:
Do you mean the real EventClient list?

Reply:
I meant we should verify a date value instead of allowing it to be null.

Reply:
due to this https://github.com/onaio/rdt-standard/pull/656#discussion_r554975610
I have set the date string explicitly and checked.